src/importexport/mei/internal/meiconverter.cpp

```diff
@@ -0,0 +1,1846 @@
+    case (engraving::DurationType::V_32ND): return libmei::DURATION_32;
+    case (engraving::DurationType::V_64TH): return libmei::DURATION_64;
+    case (engraving::DurationType::V_128TH): return libmei::DURATION_128;
+    case (engraving::DurationType::V_256TH): return libmei::DURATION_256;
+    case (engraving::DurationType::V_512TH): return libmei::DURATION_512;
+    default:
+        return libmei::DURATION_4;
+    }
+}
+
+void Convert::dynamFromMEI(engraving::Dynamic* dynamic, const StringList& meiLines, const libmei::Dynam& meiDynam, bool& warning)
+{
+    // The letters in the MEI to be mapped to SMuFL symbols in the xmlText
+    static std::map<Char, engraving::SymId> dynMap = {
+        { 'p', engraving::SymId::dynamicPiano },
+        { 'm', engraving::SymId::dynamicMezzo },
+        { 'f', engraving::SymId::dynamicForte },
+        { 'r', engraving::SymId::dynamicRinforzando },
+        { 's', engraving::SymId::dynamicSforzando },
+        { 'z', engraving::SymId::dynamicZ },
+        { 'n', engraving::SymId::dynamicNiente },
+    };
+
+    assert(dynamic);
+
+    warning = false;
+
+    // @place
+    if (meiDynam.HasPlace()) {
+        dynamic->setPlacement(meiDynam.GetPlace() == libmei::STAFFREL_above ? engraving::PlacementV::ABOVE : engraving::PlacementV::BELOW);
+        dynamic->setPropertyFlags(engraving::Pid::PLACEMENT, engraving::PropertyFlags::UNSTYLED);
+    }
+
+    // @label
+    if (meiDynam.HasLabel()) {
+        dynamic->setDynamicType(String::fromStdString(meiDynam.GetLabel()));
+    }
+    // If no @label, try to use the text
+    else if (meiLines.size() > 0 && !meiLines.at(0).contains(' ')) {
+        dynamic->setDynamicType(meiLines.at(0));
+    }
+
+    // text content
+    StringList lines;
+    // For each line in the dynamic text
+    for (const String& meiline : meiLines) {
+        String line;
+        StringList words = meiline.split(u" ");
+        bool isFirst = true;
+        // For each word in the line
+        for (const String& word : words) {
+            if (!isFirst) {
+                line += u" ";
+            }
+            // If the word is only dynamic letters, convert them to SMuFL symbols one by one
+            if (word.toStdString().find_first_not_of("fpmrszn") == std::string::npos) {
+                for (size_t i = 0; i < word.size(); i++) {
+                    line += u"<sym>" + String::fromAscii(engraving::SymNames::nameForSymId(dynMap.at(word.at(i))).ascii()) + u"</sym>";
+                }
+            }
+            // Otherwise keep it as is
+            else {
+                line += word;
+            }
+            isFirst = false;
+        }
+        lines.push_back(line);
+    }
+    dynamic->setXmlText(lines.join(u"\n"));
+
+    return;
+}
+
+libmei::Dynam Convert::dynamToMEI(const engraving::Dynamic* dynamic, StringList& meiLines)
+{
+    // The SMuFL unicode points in the plainText to be mapped to letters in the MEI
+    static std::map<char16_t, Char> dynMap = {
```

Comment:
I recommend using `static const` for such cases, and it is desirable to name the constants in uppercase, according to our code style https://github.com/musescore/MuseScore/wiki/CodeGuidelines 

```
static const std::map<char16_t, Char> DYN_MAP = {
...
```